flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/ResultPartitionWriter.java

```diff
@@ -31,12 +37,26 @@
 
 	BufferProvider getBufferProvider();
 
+	BufferPool getBufferPool();
```

Comment:
Regarding other methods:

`getNumberOfSubpartitions` and `getNumTargetKeyGroups` look like some general piece of information about partition which should be already known without `ShuffleService` from RPDD. Maybe `Environment.getWriter` should return some wrapping general class `ResultPartition` which could contain common info like those methods and `ResultPartitionWriter` produced by `ShuffleService`.

Also `getPartitionId` and `getPartitionType` look to be called only in internals of `NetworkEnviroment` and anyways available in RPDD for `Task`.

`getAllSubPartitions` is used in `ResultPartitionMetrics` which might be also specifics of `NetworkEnviroment`.

`createSubpartitionView` and `release` (used in `ResultPartitionManager`): 
Should `ResultPartitionManager` actually inside `NetworkEnviroment` (future `ShuffleService`)? It seems to be exactly at least one of the main roles of `ShuffleService` and its `releasePartitionsProducedBy` and `shutdown` might be rather associated with `ResultPartitionWriter.close` and `ShuffleService.close`.

`addBufferConsumer`, `flush`, `flushAll`, `finish` look like important methods of `ResultPartitionWriter` interface used in `RecordWriter` and `Task`.

Reply:
In general I agree with the above points of retaining only most basic methods in `ResultPartitionWriter` such as `addBufferConsumer`, `flush`, `finish`, etc, and remove other methods by refactoring existing processes or belonging to special `ShuffleService` implementations.

The initial motivation of this PR is to prepare the work for #7549 in order to make `ResultPartitionWriterWrapper` contain `ResultPartitionWriter` instead of specific `ResultPartition`.  Now I realize it is not the right way to make such temporary interface like this and refactor again after introducing the `ShuffleService` finally.

I reviewed the #7549 again and found another way to bring `ResultPartitionWriterWrapper` separately, not changing the existing methods in `ResultPartitionWriter`.  
Then the whole refactor work of `ResultPartitionWriter` seems more proper to be done after adding the `ShuffleService`, or we might refactor the interface step by step to narrow down the issues each time, such as make the `RecordWriter` not to call `ResultPartitionWriter#getBufferProvider` as a separate step, etc. What do you think?